registration-checker-compiler-plugin/src/main/scala/org/virtuslab/ash/SerializerCheckCompilerPluginComponent.scala

```diff
@@ -27,6 +28,7 @@ class SerializerCheckCompilerPluginComponent(
     s"Checks marked serializer for references to classes found in $serializerCheckPhaseName"
 
   private val serializerType = typeOf[Serializer]
+  private val serializerRegexFilter = typeOf[SerializerRegexFilter]
   private val serializabilityTraitType = typeOf[SerializabilityTrait]
 
   private var typesNotDumped = true
```

Comment:
Also, I'm getting a false positive on the current state of Hydra PR \#1190 (commit `c730950a2`) -- getting `<some-class> not referenced in marked serializer` despite it's located in line 72... oh, maybe the regex should have some extra `.*`s...

Reply:
Yeah, works with `.*Register.*`... but not with `Register.*`... does it match the entire source code lines?...

Reply:
It doesn't match source code lines, but `String` representation of `Type`. So `Register` is actually a `org.virtuslab.ash.circe.Register[A]`

Reply:
Oh okay... if you could replace `.*Option.*` in the sample test with `scala\.Option\[.*\]` maybe, this will give more clarity (other than the docs ;) )

Reply:
Also! maybe instead of having a separate optional annotation... let's make the regex a param of `@Serializer` annotation? Since without specifying the regex, the user is very likely to get a false negative just as we did on Hydra

Reply:
Huh am I missing something? I've tried

```scala
@SerializerRegexFilter("org\\.virtuslab\\.ash\\.circe\\.Register\\[.*\\]")
```

but it fails... although `"org\\.virtuslab\\.ash\\.circe\\.Register.*" passes. 

`Register` is an object so I'd expect the type to be sth like `org.virtuslab.ash.circe.Register$` (??)

Reply:
I used TypeTag for detection, it's more consistent.

Reply:
> `Register` is an object so I'd expect the type to be sth like `org.virtuslab.ash.circe.Register$` (??)

We detect types, not function calls. What gets detected is `org.virtuslab.ash.circe.Register.Registration[hydra.something.SomethingCommand]`